repose-aggregator/core/repose-core/src/main/scala/org/openrepose/core/opentracing/ReposeTracer.scala

```diff
@@ -28,14 +28,14 @@ import javax.annotation.PostConstruct
 import javax.inject.Named
 
 /**
-  * This is the Global Tracer for Repose and is based on the io.opentracing.util.GlobalTracer.
+  * This is the Global Tracer for Repose and is based on the [[io.opentracing.util.GlobalTracer]].
   *
   * The main differences are this Tracer:
   *  - immediately registers with the OpenTracing GlobalTracer
   *  - allows for registration of a new Tracer at any time.
   */
 @Named
-class ReposeTracer extends Tracer with LazyLogging {
+class ReposeTracer extends DelegatingTracer with LazyLogging {
 
   private var tracer: Tracer = NoopTracerFactory.create()
```

Comment:
I know this isn't part of this PR, but there was some discussion about replacing the default `NoopTracer` with a more fully-featured `Tracer` that simply does not report trace data (i.e., is configured with a `NoopReporter`).

Doing so, however, would probably mean keeping the configured header name in the system model so that it could be configured for use by this default `Tracer` as well. Not a big deal, just something to note. The service would have to listen for system model updates again, but that also opens the way to making this service a node service and handling it in the system model services list if we want.

Tangentially, I was under the impression that the `enabled` attributed on the service was dropped since a constant sampling could be configured to off. However, wouldn't that still lead to trace data being reported, just not exposed (perhaps even aggregated) via the tracing server? That makes me think we should either add the `enabled` attribute back, or if we switch to a node service, watch the system model to check if this service is listed.

Tangentially, does the Open Tracing Service actually needs to be a core service anymore? The `ReposeTracer` is always available without the service, the service just acts as a way to configure it. Making the service a node service would allow per-cluster configuration, and

Reply:
It doesn't work as a node service, because they'd try to replace each others tracer. We could maybe make it work by moving the `ReposeTracer` down to a lower level, but then we run into problems with the http client.